f5/bigip/tm/security/nat.py

```diff
@@ -0,0 +1,171 @@
+    ``tm:security:nat:*``
+"""
+from f5.bigip.mixins import CheckExistenceMixin
+from f5.bigip.resource import Collection
+from f5.bigip.resource import OrganizingCollection
+from f5.bigip.resource import Resource
+
+
+class Nat(OrganizingCollection):
+    """BIG-IP® AFM® Nat organizing collection."""
+
+    def __init__(self, security):
+        super(Nat, self).__init__(security)
+        self._meta_data['allowed_lazy_attributes'] = [
+            Policy_s,
+            Source_Translations,
+            Destination_Translations]
+
+
+class Rules_s(Collection):
+    """BIG-IP® AFM® Nat Rules sub-collection."""
+
+    def __init__(self, policy):
+        super(Rules_s, self).__init__(policy)
+        self._meta_data['required_json_kind'] = \
+            'tm:security:nat:policy:rules:rulescollectionstate'
+        self._meta_data['allowed_lazy_attributes'] = [Rule]
+        self._meta_data['attribute_registry'] = \
+            {'tm:security:nat:policy:rules:rulesstate':
+                Rule}
+
+
+class Rule(Resource, CheckExistenceMixin):
+    """BIG-IP® AFM® Rule resource.
+
+
+    NOTE:: The 'place-before' and 'place-after' attribute are
+        mandatory but cannot be present with one another. Those attributes
+        will not be visible when the class is created, they exist for the
+        sole purpose of rule ordering in the BIGIP. The ordering of the
+        rules corresponds to the index in the 'items' of the Rules_s
+        sub-collection.
+    """
+
+    def __init__(self, rules_s):
+        super(Rule, self).__init__(rules_s)
+        self._meta_data['required_json_kind'] = \
+            'tm:security:nat:policy:rules:rulesstate'
+
+        self._meta_data['exclusive_attributes'].append(
+            ('place-after', 'place-before'))
+        self._meta_data['minimum_additional_parameters'] = {'place-before',
+                                                            'place-after'}
+        self.tmos_ver = self._meta_data['bigip'].tmos_version
+
+    def update(self, **kwargs):
+        """We need to implement the custom exclusive parameter check."""
+        self._check_exclusive_parameters(**kwargs)
+        return super(Rule, self)._update(**kwargs)
+
+    def modify(self, **kwargs):
+        """We need to implement the custom exclusive parameter check."""
+        self._check_exclusive_parameters(**kwargs)
+        return super(Rule, self)._modify(**kwargs)
+
+    def load(self, **kwargs):
+        return super(Rule, self)._load(**kwargs)
+
+    def delete(self, **kwargs):
+        return super(Rule, self)._delete(**kwargs)
+
+    def exists(self, **kwargs):
+        return super(Rule, self)._load(**kwargs)
+
+
+class Source_Translations(Collection):
+    """BIG-IP® AFM® Nat Source Translation collection"""
+
+    def __init__(self, nat):
+        super(Source_Translations, self).__init__(nat)
+        self._meta_data['allowed_lazy_attributes'] = [Source_Translation]
+        self._meta_data['minimum_version'] = '12.1.0'
```

Comment:
From what I see NAT endpoint in its entirety is not in before 12.1.x, remove this please, we will define the version in Nat OC